pmlc/conversion/pxa_to_affine/pxa_to_affine.cc

```diff
@@ -122,6 +121,56 @@ struct AffineIfOpConversion : public OpConversionPattern<AffineIfOp> {
   }
 };
 
+static Value createReduction(ConversionPatternRewriter &rewriter,
+                             mlir::Location loc, AtomicRMWKind agg,
+                             Value source, Value val) {
```

Comment:
The variable name `val` is not very descriptive.  The calling convention is always:
```
createReduction(rewriter, op.getLoc(), op.agg(), source.getResult(), op.val());
```
Are we ever going to get location, aggregation, and value from anything but `op`?  If not the previous API makes more sense to me:

```
Value createReduction(ConversionPatternRewriter &rewriter, pxa::AffineReduceOp op, Value source)
```

Reply:
We could fix this by using a template, but you might not have noticed that `op` in both usages are actually different types.

Reply:
An interface is planned to cover all of these ops that have similar properties. Once the interface is done, this and many other parts of the code can be simplified.

Reply:
I did miss that `op` was a different type.